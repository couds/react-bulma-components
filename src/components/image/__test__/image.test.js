import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Image from '..';

describe('Image component', () => {
  it('Should exist', () => {
    expect(Image).toMatchSnapshot();
  });
  it('Should have image classname', () => {
    const component = renderer.create(<Image src="http://mydomain.com/image" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should be square', () => {
    const component = renderer.create(<Image size="square" src="http://mydomain.com/image" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should be 32x32', () => {
    const component = renderer.create(<Image size={32} src="http://mydomain.com/image" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should have use default image if error encounter', () => {
    const component = shallow(<Image src="http://mydomain.com/image" fallback="http://mydomain.com/default" />);
    const image = component.find('img');
    image.simulate('error');
    expect(component.find('img').props()).toHaveProperty('src', 'http://mydomain.com/default');
  });
  it('Should update src', () => {
    const component = shallow(<Image src="http://mydomain.com/image" />);
    component.setProps({ src: 'http://mydomain.com/other' });
    expect(component.find('img').props()).toHaveProperty('src', 'http://mydomain.com/other');
  });
  it('Should NOT update src', () => {
    const component = shallow(<Image src="http://mydomain.com/image" />);
    component.setProps({ alt: 'change prop' });
    expect(component.find('img').props()).toHaveProperty('src', 'http://mydomain.com/image');
  });
});
