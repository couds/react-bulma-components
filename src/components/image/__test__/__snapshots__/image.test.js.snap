// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component Should be 32x32 1`] = `
<figure
  className="image is-32x32"
  style={Object {}}
>
  <img
    alt=""
    onError={[Function]}
    src="http://mydomain.com/image"
  />
</figure>
`;

exports[`Image component Should be square 1`] = `
<figure
  className="image is-square"
  style={Object {}}
>
  <img
    alt=""
    onError={[Function]}
    src="http://mydomain.com/image"
  />
</figure>
`;

exports[`Image component Should exist 1`] = `[Function]`;

exports[`Image component Should have image classname 1`] = `
<figure
  className="image"
  style={Object {}}
>
  <img
    alt=""
    onError={[Function]}
    src="http://mydomain.com/image"
  />
</figure>
`;
